c7n/resources/s3.py

```diff
@@ -1495,40 +1491,16 @@ def process(self, buckets):
 
     def process_bucket(self, bucket):
         s3 = bucket_client(local_session(self.manager.session_factory), bucket)
-        state = self.data.get('state', 'enable')
-        scopes = self.data.get('scope', ['All'])
         try:
-            config = s3.get_public_access_block(
-                Bucket=bucket['Name'])['PublicAccessBlockConfiguration']
-        except ClientError as e:
-            if e.response['Error']['Code'] == 'NoSuchPublicAccessBlockConfiguration':
-                # Set config to none because NoSuchPublicAccessBlockConfiguration error
-                # was returned. This is important later because it is symbolic of
-                # all blocks being set to false. See if/else statement below
-                config = None
-            else:
-                raise
-        if config:
-            if 'All' in scopes:
-                for key in config.keys():
-                    config[key] = True if state == 'enable' else False
-            else:
-                for scope in scopes:
-                    config[scope] = True if state == 'enable' else False
+            config = dict()
+            for key in self.keys:
+                config[key] = self.data.get(key, True)
             s3.put_public_access_block(
                 Bucket=bucket['Name'],
                 PublicAccessBlockConfiguration=config
             )
-        else:
-            s3.put_public_access_block(
-                Bucket=bucket['Name'],
-                PublicAccessBlockConfiguration={
-                    'BlockPublicAcls': True if state == 'enable' else False,
-                    'IgnorePublicAcls': True if state == 'enable' else False,
-                    'BlockPublicPolicy': True if state == 'enable' else False,
-                    'RestrictPublicBuckets': True if state == 'enable' else False
-                }
-            )
+        except ClientError:
```

Comment:
if your just re-raising can omit the try/except

Reply:
Removed in 2293123